src/app/utils/memory.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Calculates the estimated base64 size in bytes based on the original file size.
+ * @param fileSizeInBytes The size of the file in bytes.
+ * @returns The estimated base64 size in bytes.
+ *
+ * Base64 encoding increases the file size by approximately 33%. It converts
+ * every 3 bytes of binary data into 4 bytes of ASCII characters. This overhead
+ * is necessary to represent binary data in a text-based format. To estimate the
+ * approximate size of the Base64 representation of a file, you can use the following formula:
+ * Base64Size ≈ (OriginalFileSize * 4) / 3
+ */
+// eslint-disable-next-line class-methods-use-this
+export function calculateBase64Size(fileSizeInBytes: number): number {
+  // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+  return Math.ceil((fileSizeInBytes * 4) / 3);
+}
+
+/**
+ * Checks if there is enough available memory to perform data encoding.
+ *
+ * @param fileSizeInBytes The size of the file in bytes.
+ * @param availableMemory The available memory in bytes.
+ * @param buffer The buffer size to add as a safety margin (default: 20% of fileSizeInBytes).
+ *               The buffer provides additional memory space to handle any unforeseen memory
+ *               requirements or fluctuations during the encoding process. It acts as a safety
+ *               margin to avoid potential out-of-memory errors. By default, the buffer size is
+ *               set to 20% of the fileSizeInBytes. You can adjust the buffer size based on your
+ *               specific memory needs. Setting a larger buffer size increases the chances of
+ *               successful encoding by allocating more memory, but it also consumes more memory
+ *               resources. Finding the right balance between buffer size and available memory
+ *               is important for optimal performance and reliability.
```

Comment:
Is this comment generated by GPT? It seems a bit over detailed. Maybe just describe what the function do and 0.2 equals additional 20% and no need to explain finding balance between buffer size and available memory is important.

Reply:
@shc261392, 
Before plan was 
- dynamically calculate user's memory (using cusotm plugin @numbersprotocol/preview-memory)
- if he can upload the file. 

However it didn't passed QA testing and QA proposed set 30 MB as constant limit after testing on several devices (asana[ comment](https://app.asana.com/0/0/1204495833338689/1204888742400415/f)). 

In #2868 there is a commit that allow 30MB file upload [fix(pre-publish): allow upload files under MAX_ALLOWED_UPLOAD_SIZE_IN…](https://github.com/numbersprotocol/capture-lite/pull/2868/commits/6ac7b81eb29aea7e33c611af095795d01ace27a5) but I forgot to clean up and remove unused code/plugin.

These commits remove unused code and plugin. Hope it resolve current conversation.
- [chore(memory.ts): remove unused method hasEnoughMemoryForEncoding](https://github.com/numbersprotocol/capture-lite/pull/2868/commits/6ecb9fef29754c8bf041ac51b6465e9d28236af1)
- [chore: remove unused plugin @numbersprotocol/preview-memory](https://github.com/numbersprotocol/capture-lite/pull/2868/commits/90642651d22f18df2017d87a7d7c7a06f8e79ecf)